zerver/openapi/zulip.yaml

```diff
@@ -10077,12 +10077,39 @@ paths:
       description: |
         Permanently delete a message.
 
-        This API corresponds to the
-        [delete a message completely][delete-completely] feature documented in
-        the Zulip help center.
+        This API corresponds to the [delete a message completely][delete-completely]
+        feature documented in the Zulip help center.
+
+        A user must be able to access the content of a message in order to delete it.
+        See [channel permissions](/help/channel-permissions) for more information
+        about content access for channel messages. For direct messages, the user
+        must have received or sent the direct message to have content access.
+
+        See [restricting message deletion](/help/restrict-message-editing-and-deletion)
+        for documentation on when users are allowed to delete messages.
+
+        The relevant realm settings in the API that are related to the above linked
+        documentation on when users are allowed to delete messages are:
+
+        - `realm_can_delete_any_message_group`
+        - `realm_can_delete_own_message_group`
+        - `realm_can_set_delete_message_policy_group`
+        - `realm_message_content_delete_limit_seconds`
```

Comment:
It'd be nice to make the connection here a bit tighter between the specific elements of the semantics, and the corresponding specific names in the API.

I believe the four of these correspond to the four settings described in the section https://zulip.com/help/restrict-message-editing-and-deletion#configure-message-deletion-permissions (within the page that's linked above). So I think a way to make that connection tighter would be:
* Link to that specific section, in the first part of this sentence.
* Then put this list in the same order as the settings are mentioned there (i.e., swap the last two items here).